src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -565,6 +576,149 @@ public static void registerTESRItemStack(Item item, int metadata, Class<? extend
+
+    public static void renderLitItem(RenderItem ri, IBakedModel model, int color, ItemStack stack) 
+    {
+        BufferBuilder bufferbuilder = Tessellator.getInstance().getBuffer();
+
+        List<BakedQuad> allquads = new ArrayList<>();
+
+        for (EnumFacing enumfacing : EnumFacing.VALUES)
+        {
+            allquads.addAll(model.getQuads(null, enumfacing, 0));
+        }
+
+        allquads.addAll(model.getQuads(null, null, 0));
+
+        // Current list of consecutive quads with the same lighting
+        List<BakedQuad> segment = new ArrayList<>();
+
+        // Lighting of the current segment
+        int segmentBlockLight = -1;
+        int segmentSkyLight = -1;
+
+        for (int i = 0; i < allquads.size(); i++) 
+        {
+            BakedQuad q = allquads.get(i);
+
+            // Lighting of the current quad
+            int bl = 0;
+            int sl = 0;
+
+            // Fail-fast on ITEM, as it cannot have light data
+            if (q.getFormat() != DefaultVertexFormats.ITEM) 
+            {
+                lightGatherer.setVertexFormat(q.getFormat());
+                if (lightGatherer.hasLighting())
+                {
+                    q.pipe(lightGatherer);
+                }
+                bl = lightGatherer.blockLight;
+                sl = lightGatherer.skyLight;
+            }
+
+            // If this is a new light value, draw the segment and flush it
+            if (segmentBlockLight != bl || segmentSkyLight != sl) 
+            {
+                if (i > 0) // Make sure this isn't the first quad being processed 
+                {
+                    drawSegment(bufferbuilder, ri, color, stack, segment, segmentBlockLight, segmentSkyLight, true);
+                }
+                segmentBlockLight = bl;
+                segmentSkyLight = sl;
+            }
+
+            segment.add(q);
+        }
+        
+        boolean lightDirty = segmentBlockLight > 0 && segmentSkyLight > 0;
+        drawSegment(bufferbuilder, ri, color, stack, segment, segmentBlockLight, segmentSkyLight, lightDirty);
+
+        // Clean up render state if necessary
+        if (lightDirty)
+        {
+            OpenGlHelper.setLightmapTextureCoords(OpenGlHelper.lightmapTexUnit, OpenGlHelper.lastBrightnessX, OpenGlHelper.lastBrightnessY);
+            GL11.glMaterial(GL11.GL_FRONT_AND_BACK, GL11.GL_EMISSION, RenderHelper.setColorBuffer(0, 0, 0, 1));
+        }
+    }
+
+    private static void drawSegment(BufferBuilder bufferbuilder, RenderItem ri, int color, ItemStack stack, List<BakedQuad> segment, int bl, int sl, boolean updateLighting) 
+    {
+        bufferbuilder.begin(GL11.GL_QUADS, DefaultVertexFormats.ITEM);
+
+        final float emissive = sl / 15f;
+        GL11.glMaterial(GL11.GL_FRONT_AND_BACK, GL11.GL_EMISSION, RenderHelper.setColorBuffer(emissive, emissive, emissive, 1));
+
+        bl *= 16;
+        sl *= 16;
```

Comment:
Is there a particular reason this multiplication is not done in the one place where these variables are used? It is a bit odd to read especially since this is mutating parameters.

Reply:
Yeah good call, I just factored in the *16 at the root calcuations.